### PR TITLE
feat: add commitizen cli and commit standards

### PR DESCRIPTION
**Pull request by kenibrewer**
# Description

Commitizen pre-installed cli will aid in the generation of commit messages can be used for automatic changelog generation.

Address #333

## What is the nature of your change?

- [x] Enhancement (adds functionality).
- [x] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (d33bs) [inline CONTRIBUTING.md:184]:
```suggestion
A pre-push commit hook from [pre-commit](https://pre-commit.com/) may be used to automatically ensure your commit messages follow this standard.
```

File: CONTRIBUTING.md
Comment (d33bs) [inline CONTRIBUTING.md:183]: Minor suggestion to move away from subjective language which might change / have multiple perspectives.
```suggestion
We prepare commit messages that follow this standard using [commitizen](https://commitizen-tools.github.io/commitizen/), which comes pre-installed in the devcontainer.
```

File: CONTRIBUTING.md
Comment (d33bs) [inline CONTRIBUTING.md:191]: Suggestion for simplifications.
```suggestion
The devcontainer will install all required formatting tools.
Formatting checks in devcontainers run automatically on any modified files before commits (using a tool called [pre-commit](https://pre-commit.com/)).
```

File: CONTRIBUTING.md
Comment (d33bs) [inline CONTRIBUTING.md:192]: What happens if someone isn't using the devcontainer? Should we make a special mention here to make sure they have an equitable chance to automatically format things with a small mention here?

File: .pre-commit-config.yaml
Comment (d33bs) [inline .pre-commit-config.yaml:23]: Reading this line and then thinking about commitzen made me wonder: are we leaving commitizen formatting of commits as a flexible option when it comes to making commits? For ex.:
- What would happen to PR's which don't abide this (would we disapprove solely based on the commit style)?
- Thinking about the end of the PR and getting ready to merge: how are squashes treated with this (would we lose the ability to leverage GitHub's built-in functionality here)?

**Review by d33bs — COMMENTED** [review]
Nice work! These were thoughtful additions and appreciate the additional context in the PR description. I left a few comments which I hope to get your feedback on before an approval. Mostly here I wonder / worry about what happens with squashes and those who might be new to the procedures (and also might be developing locally instead of in the devcontainer).

File: .pre-commit-config.yaml
Comment (kenibrewer) [reply .pre-commit-config.yaml:23]: These are good points that I think merit further discussion. I'm going to remove this pre-commit hook for now and we can add it after 1.0 release.

File: CONTRIBUTING.md
Comment (kenibrewer) [reply CONTRIBUTING.md:191]: Accepted.

File: CONTRIBUTING.md
Comment (kenibrewer) [reply CONTRIBUTING.md:183]: Accepted

File: CONTRIBUTING.md
Comment (kenibrewer) [reply CONTRIBUTING.md:184]: Removing the text about the pre-commit hook so we can discuss it further after 1.0 release.

File: CONTRIBUTING.md
Comment (kenibrewer) [reply CONTRIBUTING.md:192]: Good point. I rewrote the suggestion to be agnostic to dev environment.

File: CONTRIBUTING.md
Comment (kenibrewer) [reply CONTRIBUTING.md]: Revised this language to be dev environment agnostic.

**Comment by kenibrewer** [comment]
@d33bs I made the following changes since your last review:

- Went back to installing commitizen via poetry dev group. It turns out the previously used dev container feature is from an identically named (but separate) commitzen project from the js community. I took a closer look at the version restrictions I was worried about, but the relevant packages only moved back a few months from the latest version. Commitizen is also under active development so I'm less worried about the impacts. 
- Removed the commitizen precommit hook for further discussion and possible integration post 1.0. 
- Modifed the language about formatting to be dev environment agnostic.

**Review by d33bs — APPROVED** [review]
Thanks for the replies and follow up comments - LGTM.

**Pull request merged**
